src/vehicle_move.cpp

```diff
@@ -564,9 +566,12 @@ veh_collision vehicle::part_collision( int part, const tripoint &p,
         }
     }
 
+    if( fake_part( ret.part ) ) {
+        ret.part = fake_mounts[parts[ret.part].mount].structural_part;
+    }
     // Damage armor before damaging any other parts
     // Actually target, not just damage - spiked plating will "hit back", for example
-    const int armor_part = part_with_feature( ret.part, VPFLAG_ARMOR, true );
+    int armor_part = part_with_feature( ret.part, VPFLAG_ARMOR, true );
```

Comment:
What's the reason for this change? You didn't change the rest of the function, so it seems to be working with the const just fine.